### ISSUE TITLE
Enforce using `.setdefault()`

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1085]
# Rule request

## Thesis

One can write this code:

```python

if "list" not in dictionary:
    dictionary["list"] = []

dictionary["list"].append("list_item")
```

But, it is an anti-pattern. Instead one should write:

```python
dictionary = {}

dictionary.setdefault("list", []).append("list_item")
```

Related: https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_setdefault_to_initialize_a_dictionary.html

This is a refactoring rule.

**Comment by adambenali** [comment]
Should this be closed for the same reasons as #1084 ?

**Comment by sobolevn** [comment]
Yes, thank you!

**Comment by sobolevn** [comment]
Or should it? Let's leave it open for now. I will have to make a small research.

**Comment by ruwaid4** [comment]
@sobolevn Should this rule be implemented? If not, could you recommend us an issue to work on? We have looked at most of them but aren't up to date on what should be implemented or not.

**Comment by sobolevn** [comment]
@ruwaid4 here you go:

- https://github.com/wemake-services/wemake-python-styleguide/issues/864
- https://github.com/wemake-services/wemake-python-styleguide/issues/1025
- https://github.com/wemake-services/wemake-python-styleguide/issues/1239
